### DISCUSSION TITLE
jdtls (java lsp) creates `.classpath`, `.project` and `.settings/*` files at project root dir whenever open lazyVim

### DISCUSSION THREAD

**Discussion by devgony** [LazyVim/LazyVim#4972] [Q&A]
I just installed LazyVim and installed jdtls by `:LazyExtras`.
But Whenever open lazyVim, it creates `.classpath`, `.project` and `.settings/*` files at project root dir.

How and where to fix this?

**Comment by micah-svenson** [comment]
You need to pass the following as an argument to jdtls.
```
--jvm-arg=-Djava.import.generatesMetadataFilesAtProjectRoot=false
```

I fixed it by adding the argument in the `cmd` configuration of nvim-jdtls.

```lua
-- nvim/plugins/java.lua
return {
  {
    "mfussenegger/nvim-jdtls",
    opts = function(_, opts)
      table.insert(opts.cmd, "--jvm-arg=-Djava.import.generatesMetadataFilesAtProjectRoot=false")
    end,
  },
}
```

**Reply by Skylli202** [reply]
Thanks @micah-svenson. I've added this, but this did not move the files in `.settings/*` any clue? Thanks again for sharing your configuration snipppet :)

**Reply by micah-svenson** [reply]
You will need to manually delete the unwanted files and folders. This configuration will then prevent them from being generated again.
